types/src/data.rs

```diff
@@ -750,27 +764,54 @@ pub fn random_commitment<S: Committable>(rng: &mut dyn rand::RngCore) -> Commitm
         .finalize()
 }
 
-impl<TYPES: NodeType> Committable for ValidatingLeaf<TYPES> {
-    fn commit(&self) -> commit::Commitment<Self> {
-        let mut signatures_bytes = vec![];
-        let signatures = match &self.justify_qc.signatures {
-            YesNoSignature::Yes(signatures) => {
+/// Serialization for the QC assembled signature
+pub fn serialize_signature(signature: &AssembledSignature) -> Vec<u8> {
+    let mut signatures_bytes = vec![];
+    let signatures: Option<
+        (<BLSOverBN254CurveSignatureScheme as SignatureScheme>::Signature,
+            <BitvectorQuorumCertificate<BLSOverBN254CurveSignatureScheme> as
+            QuorumCertificateValidation<BLSOverBN254CurveSignatureScheme>>::Proof)>  = match &signature {
+            AssembledSignature::DA(signatures) => {
+                signatures_bytes.extend("DA".as_bytes());
+                Some(signatures.clone())
+            }
+            AssembledSignature::Yes(signatures) => {
                 signatures_bytes.extend("Yes".as_bytes());
-                signatures
+                Some(signatures.clone())
             }
-            YesNoSignature::No(signatures) => {
+            AssembledSignature::No(signatures) => {
                 signatures_bytes.extend("No".as_bytes());
-                signatures
+                Some(signatures.clone())
+            }
+            AssembledSignature::Genesis() => {
+                None
             }
         };
-        for (k, v) in signatures {
-            signatures_bytes.extend(&k.0);
-            signatures_bytes.extend(&v.0 .0);
-            signatures_bytes.extend(&v.1.as_bytes());
-            signatures_bytes.extend::<&[u8]>(v.2.commit().as_ref());
-        }
-        commit::RawCommitmentBuilder::new("Leaf Comm")
-            .u64_field("view_number", *self.view_number)
+    if signatures != None {
+        let (sig, proof) = signatures.unwrap();
+        let proof_bytes = bincode_opts()
+            .serialize(&proof.as_bitslice())
+            .expect("This serialization shouldn't be able to fail");
```

Comment:
Slightly concerned this might panick too. +1 for the detailed `expect`, we'll know if it fails!

Reply:
Changed to
`let (sig, proof) = signatures.
            expect("Deserialization on (sig, proof) shouldn't be able to fail.");`

Reply:
Yeah, should pay attention to all these kinds of serialization/deserialization/unwrap/...